components/tikv_util/src/codec/bytes.rs

```diff
@@ -278,6 +278,20 @@ pub fn decode_bytes_in_place(data: &mut Vec<u8>, desc: bool) -> Result<()> {
     }
 }
 
+// Returns the decoded key length of the given encoded key which is assumed valid without actually
+// decoding it.
+pub fn unchecked_decoded_len(encoded: &[u8], desc: bool) -> usize {
```

Comment:
Maybe it's fine to just use the length, since we will at most have a 1/8 * length + 7 inaccuracy.